src/IO.js

```diff
@@ -21,6 +23,17 @@ IO.prototype.chain = function(f) {
   });
 };
 
+//chainRec
+IO.chainRec = IO.prototype.chainRec = function(f, i) {
+  return new IO(function() {
+    var state = util.chainRecNext(i);
+    while (state.done === false) {
+      state = f(util.chainRecNext, util.chainRecDone, state.value).fn();
```

Comment:
Likewise here, we could use `Left` and `Right` instead of `chainRecNext` and `chainRecDone`, along with `Either.isLeft(state)` instead of `state.done === false`.